tarantool/connection.py

```diff
@@ -767,12 +769,30 @@ def _send_request_wo_reconnect(self, request):
 
         assert isinstance(request, Request)
 
+        # Flag for detecting the last message as out-of-band.
+        iproto_chunk_detected = False
+        
         response = None
         while True:
             try:
-                self._socket.sendall(bytes(request))
+                if not iproto_chunk_detected:
+                    # If the last received message is out-of-band, 
+                    # the request will not be sent.
+                    self._socket.sendall(bytes(request))
                 response = request.response_class(self, self._read_response())
-                break
+                if response._code == IPROTO_CHUNK:
+                    # Сase of receiving an out-of-band message.
+                    pushed_data.append(copy.deepcopy(response._data))
```

Comment:
As far as I understand, there is no pushed_data in Lua API. I'm not sure we should add it as well. If one would want to process it or collect it, he always may use `on_push` and `on_push_res`.

Reply:
Oh, there is no `on_push_res` in Lua as well. I think it's better to use exactly the same api with `on_push` and `on_push_ctx` with `None` default values for both.

This way, we won't add any extra load for request which wants to ignore possible push messages (except for `if callable()` check), yet we will support all possible cases (including "collect all messages to a list"). Then we just provide a couple of small examples in doc about how to work with iproto.

Reply:
Done, also updated the PR description in this part